src/libmscore/score.cpp

```diff
@@ -3418,7 +3418,7 @@ void Score::selectRange(Element* e, int staffIdx)
                 Segment* s1 = tick2segmentMM(t1, true, SegmentType::ChordRest);
                 Segment* s2 = tick2segmentMM(t2, true, SegmentType::ChordRest);
                 if (s2) {
-                    s2 = s2->next1MM(SegmentType::ChordRest);
+                    s2 = s2->next1MM(SegmentType::ChordRest | SegmentType::EndBarLine);
```

Comment:
Nice job tracking down where this goes wrong!  But I wonder, maybe this isn't the best solution.  Elsewhere we use the convention of passing nullptr as the end segment when creating a range, as the way of saying "select to end of score".  Maybe that's a better answer here - just let s2 be nullptr if that's what next1MM returns, and change the next line to simply "if (s1)"?

My only hesitation about recommending this is that elsewhere there are some odd inconsistencies with the end of selection where sometimes the end barline is selected and sometimes not (this is true for selections within the score too, not just at the end).  So maybe for MuseScore 4 we want to consider the best way to handle this.

Reply:
Tested:

                        if (s2)
                              s2 = s2->next1MM(SegmentType::ChordRest);
                        if (s1) {
                              _selection.setRange(s1, s2, idx1, idx2 + 1);

Indeed works too. And deleting code (here: the `& s2`) to fix a bug seems nicer than adding some ;-)

Reply:
Thanks for the detailed feedback :). I've seen nullptr being used in other conditionals of score.cpp but didn't notice that it can also be an end marker itself. Indeed this looks like a more elegant fix to me -- made a new commit with the suggested changes.